src/databricks_ai_bridge/genie.py

```diff
@@ -75,24 +93,34 @@ def create_message(self, conversation_id, content):
 
     def poll_for_result(self, conversation_id, message_id):
         def poll_result():
-            while True:
+            iteration_count = 0
+            while True and iteration_count < MAX_ITERATIONS:
+                iteration_count += 1
                 resp = self.genie._api.do(
                     "GET",
                     f"/api/2.0/genie/spaces/{self.space_id}/conversations/{conversation_id}/messages/{message_id}",
                     headers=self.headers,
                 )
                 if resp["status"] == "EXECUTING_QUERY":
-                    sql = next(r for r in resp["attachments"] if "query" in r)["query"]["query"]
+                    query = next(r for r in resp["attachments"] if "query" in r)["query"]
+                    description = query.get("description", "")
+                    sql = query.get("query", "")
+                    logging.debug(f"Description: {description}")
                     logging.debug(f"SQL: {sql}")
                     return poll_query_results()
                 elif resp["status"] == "COMPLETED":
                     return next(r for r in resp["attachments"] if "text" in r)["text"]["content"]
+                elif resp["status"] == "FAILED":
```

Comment:
There are a few more message statuses according to the sdk. Should we also add `CANCELLED` and `QUERY_RESULT_EXPIERED` here? 

https://databricks-sdk-py.readthedocs.io/en/latest/dbdataclasses/dashboards.html#databricks.sdk.service.dashboards.MessageStatus